src/app/users/users.states.js

```diff
@@ -7,12 +7,15 @@ angular.module('users')
       parent: 'index',
       url: '/users',
       templateUrl: 'app/users/users.html',
-      controller: 'UsersCtrl',
-      controllerAs: 'usersCtrl',
+      controller: function ($state) {
+        if ($state.current.name === 'users') {
+          $state.go('users.all')
```

Comment:
This shouldn't be necessary; to set a default state, you define the parent abstract and set an empty url of a child (`url: ''`). We already have this in `users` and `users.all`. Is this not working for you? The downside of doing it via a controller is the user can see a flash of the parent state's view before the child has resolved.